### PR TITLE
Fix spiller types num

### PR DESCRIPTION
**Pull request by duanmeng**

### CODE REVIEW

File: velox/exec/tests/SpillerTest.cpp [hunk 1]
```diff
@@ -70,11 +70,12 @@ struct TestParam {
 struct TestParamsBuilder {
   std::vector<TestParam> getTestParams() {
     std::vector<TestParam> params;
-    for (int i = 0; i < Spiller::kNumTypes; ++i) {
+    auto spillerTypes = static_cast<int8_t>(Spiller::Type::kNumTypes);
```
Comment (xiaoxmeng) [inline velox/exec/tests/SpillerTest.cpp:73]:
```
const auto numSpillerTypes = 
```

**Review by xiaoxmeng — COMMENTED** [review]
@duanmeng thanks for the fix % nit.

File: velox/exec/Spiller.h [hunk 2]
```diff
@@ -37,8 +37,9 @@ class Spiller {
     kHashJoinProbe = 3,
     // Used for order by.
     kOrderBy = 4,
+    // Num of spiller types
```
Comment (xiaoxmeng) [inline velox/exec/Spiller.h:40]: nit: Number of ...

**Review by xiaoxmeng — APPROVED** [review]
@duanmeng LGTM. Thanks!

File: velox/exec/Spiller.h (hunk 2 above)
Comment (duanmeng) [reply velox/exec/Spiller.h]: Fixed, and add `.` at the end of this comment line.

File: velox/exec/Spiller.h [hunk 3]
```diff
@@ -37,8 +37,9 @@ class Spiller {
     kHashJoinProbe = 3,
     // Used for order by.
     kOrderBy = 4,
+    // Number of spiller types.
+    kNumTypes = 5,
```
Comment (amitkdutta) [inline velox/exec/Spiller.h:41]: @duanmeng @xiaoxmeng Curious, whats the issue with previous code (except for Type : int8_t)? Previous code looks more readable.

Comment (xiaoxmeng) [reply velox/exec/Spiller.h:41]: @amitkdutta previous code kNumTypes is 4 so some test can't cover. By putting kNumTypes into enum, we can kind of preventing this.

**Pull request closed without merging**